src/bin/ion/commands/generate/model.rs

```diff
@@ -9,46 +10,60 @@ use std::collections::HashMap;
 // _Note: This model will eventually use a map (FullQualifiedTypeReference, DataModel) to resolve some the references in container types(sequence or structure)._
 // TODO: This is not yet used in the implementation, modify current implementation to use this data model.
 use crate::commands::generate::context::SequenceType;
+use crate::commands::generate::result::CodeGenResult;
 use serde::Serialize;
+use serde_json::Value;
 
 /// Represent a node in the data model tree of the generated code.
 /// Each node in this tree could either be a module/package or a concrete data structure(class, struct, enum etc.).
 /// This tree structure will be used by code generator and templates to render the generated code as per given ISL type definition hierarchy.
-#[allow(dead_code)]
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, PartialEq, Serialize)]
 pub struct DataModelNode {
     // Represents the name of this data model
     // Note: It doesn't point to the fully qualified name. To get fully qualified name use `fully_qualified_name()` from `AbstractDataType`.
-    name: String,
+    pub(crate) name: String,
     // Represents the type of the data model
     // It can be `None` for modules or packages.
-    code_gen_type: Option<AbstractDataType>,
+    pub(crate) code_gen_type: Option<AbstractDataType>,
     // Represents the nested types for this data model
-    nested_types: Vec<DataModelNode>,
+    pub(crate) nested_types: Vec<DataModelNode>,
 }
 
 impl DataModelNode {
-    #![allow(dead_code)]
+    #[allow(dead_code)]
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    #[allow(dead_code)]
     pub fn is_scalar(&self) -> bool {
         if let Some(code_gen_type) = &self.code_gen_type {
             return matches!(code_gen_type, AbstractDataType::Scalar(_));
         }
         false
     }
 
+    #[allow(dead_code)]
     pub fn is_sequence(&self) -> bool {
         if let Some(code_gen_type) = &self.code_gen_type {
             return matches!(code_gen_type, AbstractDataType::Sequence(_));
         }
         false
     }
 
+    #[allow(dead_code)]
     pub fn is_structure(&self) -> bool {
         if let Some(code_gen_type) = &self.code_gen_type {
             return matches!(code_gen_type, AbstractDataType::Structure(_));
         }
         false
     }
+
+    pub fn fully_qualified_type_ref(&mut self) -> Option<FullyQualifiedTypeReference> {
```

Comment:
What does `None` mean here? Anonymous type?

Reply:
Even anonymous types must be given a name eventually. I think that this should probably not be `Option<_>`.

Reply:
This `None` here represents if the given ISL type is `struct`, `list` or `sexp` (container types) , since code gen doesn't support open ended types for these containers.

Reply:
I have added document comment for this.